fenix/app/src/main/java/org/mozilla/fenix/translations/DownloadIndicator.kt

```diff
@@ -0,0 +1,235 @@
+) {
+    BaseDownloadIndicator(
+        text = text,
+        textColor = textColor,
+        modifier = modifier,
+        buttonContentDescription = contentDescription,
+        icon = icon,
+        backgroundColor = backgroundColor,
+        iconModifier = iconModifier.then(
+            Modifier.wrapContentSize(),
+        ),
+        iconContentDescription = iconContentDescription,
+        iconTint = iconTint,
+    )
+}
+
+/**
+ * Icon for Download indicator.
+ *
+ * @param modifier [Modifier] to be applied to the icon layout.
+ * @param icon [Painter] used to be displayed.
+ * @param tint to be applied to the icon.
+ * @param iconContentDescription Optional contentDescription for the icon.
+ * */
+@Composable
+fun DownloadIconIndicator(
+    modifier: Modifier = Modifier,
+    icon: Painter,
+    tint: Color = FirefoxTheme.colors.iconActionPrimary,
+    iconContentDescription: String? = null,
+) {
+    BaseLoadingIndicator(
+        icon = icon,
+        modifier = modifier.then(
+            Modifier
+                .rotate(rotationAnimation())
+                .semantics {
+                    if (iconContentDescription != null) {
+                        contentDescription = iconContentDescription
+                    }
+                },
+        ),
+        tint = tint,
+    )
+}
+
+@Composable
+private fun BaseDownloadIndicator(
+    text: String,
+    textColor: Color,
+    modifier: Modifier = Modifier,
+    buttonContentDescription: String? = null,
+    icon: Painter? = null,
+    backgroundColor: Color,
+    iconModifier: Modifier = Modifier,
+    iconTint: Color,
+    iconContentDescription: String? = null,
+) {
+    // Required to detect if font increased due to accessibility.
+    val fontScale: Float = LocalConfiguration.current.fontScale
+
+    Button(
+        enabled = false,
+        onClick = { },
+        modifier = modifier.then(
+            Modifier.semantics {
+                contentDescription = if (buttonContentDescription.isNullOrEmpty()) {
+                    text
+                } else {
+                    buttonContentDescription
+                }
+            },
+        ),
+        contentPadding = PaddingValues(horizontal = 16.dp, vertical = 12.dp),
+        elevation = ButtonDefaults.elevation(defaultElevation = 0.dp, pressedElevation = 0.dp),
+        colors = ButtonDefaults.outlinedButtonColors(
+            backgroundColor = backgroundColor,
+        ),
+    ) {
+        icon?.let { painter ->
+            DownloadIconIndicator(
+                icon = painter,
+                modifier = iconModifier,
+                tint = iconTint,
+                iconContentDescription = iconContentDescription,
+            )
+            Spacer(modifier = Modifier.width(8.dp))
+        }
+
+        Text(
+            text = text,
+            textAlign = TextAlign.Center,
+            color = textColor,
+            modifier = Modifier.clearAndSetSemantics { },
```

Comment:
```suggestion
            modifier = Modifier.clearAndSetSemantics {},
```
nit: Prefer `{}` over `{ }`. Same in other places.

Reply:
Done